Crash in HttpClientTask

Reported by our client.

It looks like the connection parameter in logRespose might be null here:

https://github.com/wultra/powerauth-mobile-sdk/blob/develop/proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpClientTask.java#L308

Stacktrace:

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
    at android.os.AsyncTask$4.done(AsyncTask.java:415)
    at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
    at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
    at java.util.concurrent.FutureTask.run(FutureTask.java:269)
    at io.getlime.security.powerauth.sdk.impl.DefaultExecutorProvider$SerialExecutor$1.run(DefaultExecutorProvider.java:68)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
    at java.lang.Thread.run(Thread.java:1012)
   Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.net.URL java.net.HttpURLConnection.getURL()' on a null object reference
    at io.getlime.security.powerauth.networking.client.HttpClientTask.logResponse(HttpClientTask.java:315)
    at io.getlime.security.powerauth.networking.client.HttpClientTask.doInBackground(HttpClientTask.java:215)
    at android.os.AsyncTask$3.call(AsyncTask.java:394)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at io.getlime.security.powerauth.sdk.impl.DefaultExecutorProvider$SerialExecutor$1.run(DefaultExecutorProvider.java:68) 
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644) 
    at java.lang.Thread.run(Thread.java:1012) 
```

--- a/proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpClientTask.java
+++ b/proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpClientTask.java
@@ -269,14 +269,15 @@ class HttpClientTask<TRequest, TResponse> extends AsyncTask<TRequest, Void, TRes
      * @param connection prepared connection object.
      * @param requestData (optional) byte array with request data.
      */
-    private void logRequest(HttpURLConnection connection, byte[] requestData) {
+    private void logRequest(@Nullable HttpURLConnection connection, @Nullable byte[] requestData) {
         if (!PowerAuthLog.isEnabled()) {
             return;
         }
         // Endpoint
         final IEndpointDefinition<TResponse> endpoint = httpRequestHelper.getEndpoint();
         // URL, method
-        final String url = connection.getURL().toString();
+        final boolean hasConnection = connection != null;
+        final String url = hasConnection ? connection.getURL().toString() : "null";
         final String method = endpoint.getHttpMethod();
         // Flags
         final boolean signature = endpoint.getAuthorizationUriId() != null;
@@ -287,7 +288,7 @@ class HttpClientTask<TRequest, TResponse> extends AsyncTask<TRequest, Void, TRes
             PowerAuthLog.d("HTTP %s request%s: -> %s", method, signedEncrypted, url);
         } else {
             // Verbose, put headers and body (if not encrypted) into the log.
-            final Map<String,List<String>> prop = connection.getRequestProperties();
+            final Map<String,List<String>> prop = hasConnection ? connection.getRequestProperties() : null;
             final String propStr = prop == null ? "<empty>" : prop.toString();
             if (encrypted) {
                 PowerAuthLog.d("HTTP %s request%s: %s\n- Headers: %s- Body: <encrypted>", method, signedEncrypted, url, propStr);
@@ -305,14 +306,15 @@ class HttpClientTask<TRequest, TResponse> extends AsyncTask<TRequest, Void, TRes
      * @param responseData (optional) data returned in HTTP request.
      * @param error (optional) error produced during the request.
      */
-    private void logResponse(HttpURLConnection connection, byte[] responseData, Throwable error) {
+    private void logResponse(@Nullable HttpURLConnection connection, @Nullable byte[] responseData, @Nullable Throwable error) {
         if (!PowerAuthLog.isEnabled()) {
             return;
         }
         // Endpoint
         final IEndpointDefinition<TResponse> endpoint = httpRequestHelper.getEndpoint();
         // URL, method
-        final String url = connection.getURL().toString();
+        final boolean hasConnection = connection != null;
+        final String url = hasConnection ? connection.getURL().toString() : "null";
         final String method = endpoint.getHttpMethod();
         final String errorMessage;
         if (error != null) {
@@ -329,7 +331,7 @@ class HttpClientTask<TRequest, TResponse> extends AsyncTask<TRequest, Void, TRes
         // Response code
         int responseCode;
         try {
-            responseCode = connection.getResponseCode();
+            responseCode = hasConnection ? connection.getResponseCode() : 0;
         } catch (IOException e) {
             responseCode = 0;
         }
@@ -343,7 +345,7 @@ class HttpClientTask<TRequest, TResponse> extends AsyncTask<TRequest, Void, TRes
         } else {
             final boolean encrypted = endpoint.getEncryptorId() != EciesEncryptorId.NONE;
             // Response headers
-            final String responseHeaders = connection.getHeaderFields().toString();
+            final String responseHeaders = hasConnection ? connection.getHeaderFields().toString() : "{}";
             // Response body
             final String responseBodyTmp = responseData == null ? "<empty>" : new String(responseData, Charset.defaultCharset());
             final String responseBody;

